gtsfm/evaluation/metrics_report.py

```diff
@@ -134,13 +169,67 @@ def get_figures_for_metrics(metrics_group: GtsfmMetricsGroup) -> Tuple[str, str]
                 raise ValueError(f"Metric {metric_name} does not contain a summary.")
             # Add a scalar metric for mean of 1D distributions.
             scalar_metrics["mean_" + metric_name] = value[metrics.SUMMARY_KEY]["mean"]
+            scalar_metrics["median_" + metric_name] = value[metrics.SUMMARY_KEY]["median"]
         else:
             scalar_metrics[metric_name] = value
     table = create_table_for_scalar_metrics(scalar_metrics)
     plots_fig = create_plots_for_distributions(metrics_dict)
     return table, plots_fig
 
 
+def get_figures_for_metrics_and_compare(metrics_group: GtsfmMetricsGroup, metric_path: str) -> Tuple[str, str]:
+    """Gets the tables and plots for individual metrics in a metrics group.
+
+    All scalar metrics are reported in the table.
+    Metrics of 1-D distributions have an entry in the table for the mean,
+    and a histogram or box plot as per their property.
+
+    Args:
+        metrics_group: A GtsfmMetricsGroup for any gtsfm module.
+
+    Returns:
+        A tuple of table and plotly figures as HTML code.
+    """
+    all_scalar_metrics = []
+    all_metrics_groups = []
+
+    colmap_metric_path = metric_path[: metric_path.rindex("/")] + "/colmap" + metric_path[metric_path.rindex("/") :]
+    colmap_metrics_group = GtsfmMetricsGroup.parse_from_json(colmap_metric_path)
+
+    all_metrics_groups.append(metrics_group)
+    all_metrics_groups.append(colmap_metrics_group)
+
+    for metrics_group in all_metrics_groups:
+        scalar_metrics = {}
+        metrics_dict = metrics_group.get_metrics_as_dict()[metrics_group.name]
+        # Separate the scalar metrics.
+        for metric_name, value in metrics_dict.items():
+            if isinstance(value, dict):
+                # Metrics with a dict representation must contain a summary.
+                if metrics.SUMMARY_KEY not in value:
+                    raise ValueError(f"Metric {metric_name} does not contain a summary.")
+                # Add a scalar metric for mean of 1D distributions.
+                mean_nan = value[metrics.SUMMARY_KEY]["mean"] != value[metrics.SUMMARY_KEY]["mean"]
+                median_nan = value[metrics.SUMMARY_KEY]["median"] != value[metrics.SUMMARY_KEY]["median"]
+                if mean_nan or median_nan:
+                    scalar_metrics["mean_" + metric_name] = ""
+                    scalar_metrics["median_" + metric_name] = ""
+                else:
+                    scalar_metrics["mean_" + metric_name] = value[metrics.SUMMARY_KEY]["mean"]
+                    scalar_metrics["median_" + metric_name] = value[metrics.SUMMARY_KEY]["median"]
+            else:
+                scalar_metrics[metric_name] = value
```

Comment:
this does not guarantee that the two dicts (colmap and gtsfm) have the same keys in order. If for some reason they don't, things might fail, or even worse show incorrect results. I think `create_table_for_scalar_metrics_and_compare` must accept a Dict[str, List[Union[float, int]]], and so this function should handle the logic of merging the keys.

Reply:
Why should `create_table_for_scalar_metrics_and_compare` accept a Dict[str, List[Union[float, int]]] instead of List[Dict[str, Union[float, int]]]? `create_table_for_scalar_metrics`, which operates on a single pipeline's (GTSfM) metrics accepts a Dict[str, Union[float, int]]. Extending this to multiple pipelines for comparison led me to a list of pipeline's metrics. 

Would you be opposed to using an OrderedDict or sorting the dicts by key to ensure order?

Reply:
`Dict[str, List[Union[float, int]]]` because the metrics from multiple pipelines we compare are still the same metrics, that share the same keys. Only the value of the metric differs. 

Yeah I think I would be opposed. Using an OrderedDict may work (it is the reason its currently working because python dicts are ordered by default), but what if one of the pipelines don't output a metric? Then all the other metrics get matched incorrectly. So we need some function that takes these individual dicts and merges them into one dict. The logic for that I think is straightforward.

Reply:
I've changed to `Dict[str, List[Union[float, int]]]`. For future reference, the Dicts for other pipelines have the same keys (metrics) as GTSfM and are left empty if the corresponding metric isn't available.